### PR TITLE
Add support for gitlab/bitbucket sources

### PR DESCRIPTION
**Pull request by rgrinberg**
It's just as easy to support them as github

### CODE REVIEW

File: src/dune_rules/watermarks.ml [hunk 1]
```diff
@@ -239,8 +239,9 @@ let make_watermark_map ~commit ~version ~dune_project ~package =
     | Some value -> Ok (String.concat ~sep value)
   in
   let make_dev_repo_value = function
-    | Some (Package.Source_kind.Github (user, repo)) ->
-      Ok (sprintf "https://github.com/%s/%s" user repo)
+    | Some (Package.Source_kind.Host h) ->
+      (* Is this right? shouldn't this include git+https? *)
```
Comment (emillon) [inline src/dune_rules/watermarks.ml:243]: I think `git+https` is only necessary for `opam pin` when `-k git` is not passed. `git clone` works fine with just `https`. So it really depends what `%PKG_REPO%` is supposed to be. It's described as "the value of the repo field", but there's no repo field in opam. There's dev-repo, but it includes `git+https` in `dune.opam`, so it must be filled through another path. So I'm confused as well :)

Comment (rgrinberg) [reply src/dune_rules/watermarks.ml:243]: I see. That makes sense. I'll remove the comment then.

File: src/dune_engine/package.ml [hunk 2]
```diff
@@ -225,33 +225,94 @@ module Dependency = struct
...
 
 module Source_kind = struct
+  module Host = struct
+    type kind =
+      | Github
+      | Bitbucket
+      | Gitlab
+
+    let kind_string_map =
+      [ (Github, "github"); (Bitbucket, "bitbucket"); (Gitlab, "gitlab") ]
```
Comment (jeremiedimino) [inline src/dune_engine/package.ml:235]: This is only used as `List.assoc kind_string_map kind |> Option.value_exn`. Feels like a plain `function Github -> "github" | ...` function would be simpler and more efficient

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by jeremiedimino** [comment]
LGTM appart from the `kind_string_map`, which feels unnecessarily complicated

**Pull request merged**